### ISSUE TITLE
Handling of help messages for interactive command line options

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida_core#1683]
Since the move to click introduces some changes in interactive commands, I'd like to take this opportunity to try and agree on some basic rules for the interactive user interface.

### Before click migration

Let's look e.g. at `verdi computer setup`
```
$ verdi computer setup
At any prompt, type ? to get some help.
---------------------------------------
=> Computer name: asdfsaf
Creating new computer with name 'asdfsaf'
=> Fully-qualified hostname:
Invalid input: No hostname specified
Enter '?' for help
=> Fully-qualified hostname: ?
HELP: The fully qualified host-name of this computer
=> Fully-qualified hostname:
```

 1. the top line points out what to do in order to get help
 1. all input lines are prefixed with `=>`
 1. if user enters invalid input, the user is informed of this fact and notified how to access help
 1. in order to actually see the help, the user still needs to press `?`

### Proposal for click migration

I propose the following interface:
```
$ verdi computer setup
=> Computer name: asdfsaf
=> Fully-qualified hostname:
Info: The fully qualified host-name of this computer
=> Fully-qualified hostname: localhost
=> Description: my computer
=> Enabled [True]: 
=> Transport type: badtransport
Info: The name of the transport to be used. Valid names are: local,ssh
Error: The specified transport is not a valid one
=> Transport type:
```

 1. input lines continue to be prefixed with `=>`
 1. if the user hits return without any other input, the help string is shown
 1. if the user provides invalid input, the help string is show, plus potentially helpful information from the validator (e.g. entered path is not an absolute path, ...)

Happy to hear your feedback @muhrin @giovannipizzi 

What is currently missing:
 * prefixing input lines with `=>` (or: since we are using colors now, one could define a color for prompts instead)
 * show help message on empty and invalid input (using `echo_info`)
 * use `echo_error` for showing error message from type validators

**Comment by DropD** [comment]
Instead of removing I would suggest putting the message indicating the use of '?' back in place.

Reason to keep the '?': To get help on the meaning of an option with default, e.g.
```
Enabled [True]: ?
<helpful info about what "enabled" means>
```
Note, that just entering would accept, having to enter something invalid (having to guess what is invalid as well) would be just stupid.

**Comment by DropD** [comment]
fixed in #1684
